pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -394,6 +394,10 @@ func (gce *GCECloud) EnsureTCPLoadBalancer(name, region string, externalIP net.I
 		PortRange:  fmt.Sprintf("%d-%d", minPort, maxPort),
 		Target:     gce.targetPoolURL(name, region),
 	}
+	if loadBalancerIP != nil {
+		req.IPAddress = loadBalancerIP.String()
```

Comment:
Right, so it DOES sound like the OS balancer is ready for this feature,
just using the wrong field :)

On Tue, Sep 1, 2015 at 12:54 AM, Angus Lees <notifications@github.com>
wrote:

> In pkg/cloudprovider/providers/gce/gce.go
> <https://github.com/kubernetes/kubernetes/pull/13005#discussion_r38393572>
> :
>
> > @@ -394,6 +394,10 @@ func (gce *GCECloud) EnsureTCPLoadBalancer(name, region string, externalIP net.I
> >  		PortRange:  fmt.Sprintf("%d-%d", minPort, maxPort),
> >  		Target:     gce.targetPoolURL(name, region),
> >  	}
> > +	if loadBalancerIP != nil {
> > +		req.IPAddress = loadBalancerIP.String()
>
> It uses the externalIP arg passed to EnsureTCPLoadBalancer (if non-nil).
>
> So there's two cases (as I understand the existing code):
>
>    - type=LoadBalancer and no ExternalIPs: the plugin allocates a new
>    public IP, uses it, and destroys it when you remove the
>    service/loadbalancer.
>    - type=LoadBalancer and ExternalIPs: the plugin is passed the first
>    externalIP and the openstack plugin just uses it as the loadbalancer IP. If
>    the user provided an invalid IP for some reason, it will return an error.
>    How/where the user got that IP is up to them.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/13005/files#r38393572>.
>